src/script/descriptor.cpp

```diff
@@ -339,14 +339,15 @@ class DescriptorImpl : public Descriptor
 {
     //! Public key arguments for this descriptor (size 1 for PK, PKH, WPKH; any size for Multisig).
     const std::vector<std::unique_ptr<PubkeyProvider>> m_pubkey_args;
+    //! The string name of the descriptor function.
+    const std::string m_name;
```

Comment:
I don't mind, but why are you making `m_name` public?

Reply:
It doesn't. The default visibility is private.

You should really see this change as moving m_subdescriptor_arg into the `protected` area rather than moving `m_mame`.

Reply:
Oops, I misread that move. That makes sense.

Reply:
However there's a warning:
```
script/descriptor.cpp:368:176: warning: field 'm_subdescriptor_arg' will be initialized after field 'm_name' [-Wreorder]
    DescriptorImpl(std::vector<std::unique_ptr<PubkeyProvider>> pubkeys, std::unique_ptr<DescriptorImpl> script, const std::string& name) : m_pubkey_args(std::move(pubkeys)), m_subdescriptor_arg(std::move(script)), m_name(name) {}
```

Reply:
That should be fixed now.